src/build-defs.cmake

```diff
@@ -37,6 +37,12 @@ if(APACHE_ONLY)
   add_definitions(-DAPACHE_ONLY)
 endif()
 
+file(READ ${PG_INCLUDEDIR_SERVER}/nodes/parsenodes.h PG_PARSENODES_H)
+string(REGEX MATCH "AT_CookedColumnDefault" PG_HAS_COOKEDCOLUMNDEFAULT ${PG_CONFIG_H})
+if (PG_HAS_COOKEDCOLUMNDEFAULT)
+  add_definitions(-DPG_HAS_COOKEDCOLUMNDEFAULT)
```

Comment:
A couple of questions/suggestions:

* Shouldn't this be tied to PostgreSQL version rather than parsing header?
* I think we should write this to `config.h` rather than relying on command-line/compile-time defines. It is better to be able to see the build configuration in a single file.

Reply:
this code is not yet in a specific version thats why i check the header, but it's in the REL_12_STABLE branch and in snapshot builds which we use in our ABI tests, once 12.5 gets tagged we can turn this into a simple version check.

Reply:
For the former (tying to PG version), I guess something like this would be sufficient in a header file or in `process_utility.c`

```
#if ((PG11 && PG_VERSION_NUM > XXXXXX) || (PG12 && PG_VERSION_NUM > YYYYYY) || PG12_GT)
#define PG_HAS_COOKEDCOLUMNDEFAULT 1
#endif
```

Reply:
that code would not work, see my previous comment or commit message why i did it this way

Reply:
Got it. I recommend adding a note/comment about this in the CMake file (and maybe the commit message) since it is non-obvious why it was done this way.